lib/mbr.mli

```diff
@@ -79,11 +79,11 @@ type t = private {
   seconds : int;
   minutes : int;
   hours : int;
-  disk_signature : int32;
+  disk_signature : int32 option;
```

Comment:
This does not need changing. We want the disk signature to always be present, and in the "smart" constructor (function `make`) we fill in a default value if not provided.
```suggestion
  disk_signature : int32;
```

Reply:
Will the default value be something like `0l`?

Reply:
something like `let make (?disk_signature = 0)l in partitions`

Reply:
And change back the line `disk_signature = (match disk_signature with Some s -> s | None -> 0l);` to just `disk_signature;`

Reply:
I updated the code to reflect the changes, but i wish to say the implementation of the optional argument is what I don't understand since the way i saw it used in the documentation you sent is using the `?disk_signature`. thanks

Reply:
Yes, optional arguments in OCaml is a bit convoluted. There are many ways you can define a function with optional arguments: `let f ?x () = Option.get x`, `let f ?(x=42) () = x`, `let f ?x:(y=42) () = y`. Then there's a different syntax for calling the function, and for the type of the function. It may not have been the best very first issue, but I am glad that you have solved it!